src/tests/ftest/util/server_utils.py

```diff
@@ -50,292 +47,352 @@ class ServerFailed(Exception):
-    """
-    kill_cmds = [
-        "pkill '(daos_server|daos_io_server)' --signal INT",
-        "sleep 5",
-        "pkill '(daos_server|daos_io_server)' --signal KILL",
-    ]
-    # Intentionally ignoring the exit status of the command
-    pcmd(hosts, "; ".join(kill_cmds), False, None, None)
+    def stop(self):
+        """Stop the server through the runner."""
+        if self.runner.job.yaml.is_nvme():
+            storage_reset(self._hosts)
+
+        print("Stopping servers")
+        error_list = []
+        try:
+            self.runner.stop()
+        except CommandFailure as error:
+            error_list.append("Error stopping servers: {}".format(error))
+
+        return error_list
+
+    def prep(self):
+        """Prepare the hosts before starting daos server.
+
+        Args:
+            hosts (list): list of host names to prepare
+
+        Raises:
+            ServerFailed: if there is any errors preparing the hosts
+
+        """
+        # Kill any doas servers running on the hosts
+        self.kill()
+
+        # Clean up any files that exist on the hosts
+        self.clean(scm=True)
+
+        # Verify the domain socket directory is present and owned by this user
+        user = getpass.getuser()
+        file_checks = (
+            ("Server", self._hosts, "/var/run/daos_server"),
+        )
+        for host_type, host_list, directory in file_checks:
+            status, nodeset = check_file_exists(host_list, directory, user)
+            if not status:
+                raise ServerFailed(
+                    "{}: {} missing directory {} for user {}.".format(
+                        nodeset, host_type, directory, user))
+
+    def kill(self):
+        """Forcably kill any daos server processes running on hosts.
+
+        Sometimes stop doesn't get everything.  Really whack everything
+        with this.
+
+        """
+        kill_cmds = [
+            "pkill '(daos_server|daos_io_server)' --signal INT",
+            "sleep 5",
+            "pkill '(daos_server|daos_io_server)' --signal KILL",
+        ]
+        # Intentionally ignoring the exit status of the command
+        result = pcmd(self._hosts, "; ".join(kill_cmds), False, None, None)
+        if len(result) > 1 or 0 not in result:
+            raise ServerFailed(
+                "Error cleaning tmpfs on servers: {}".format(
+                    ", ".join(
+                        [str(result[key]) for key in result if key != 0])))
+
+    def clean(self, scm=False):
+        """Clean the tmpfs on the servers.
+
+        Args:
+            scm (bool): if true, remove tmpfs mount
+        """
+        clean_cmds = [
+            "find /mnt/daos -mindepth 1 -maxdepth 1 -print0 | xargs -0r rm -rf"
+        ]
+
+        if self.runner.job.yaml.is_scm() and scm:
+            clean_cmds.append("sudo umount /mnt/daos; sudo rm -rf /mnt/daos")
+
+        # Intentionally ignoring the exit status of the command
```

Comment:
Again, are you really ignoring the exit status?  Isn't lines 390-395 below looking at the exit statuses?

Reply:
Took those lines out as well.

Reply:
To finish resolving this the comment should also be removed